src/commands/contract/inspect_contract/mod.rs

```diff
@@ -0,0 +1,456 @@
+    let contract_code_view =
+        if let near_jsonrpc_primitives::types::query::QueryResponseKind::ViewCode(result) =
+            view_code_response.kind
+        {
+            result
+        } else {
+            return Err(color_eyre::Report::msg("Error call result".to_string()));
+        };
+
+    let account_view = get_account_view(
+        &network_config.network_name,
+        &json_rpc_client,
+        &block_reference,
+        account_id,
+    )
+    .await?;
+
+    let access_keys = get_access_keys(
+        &network_config.network_name,
+        &json_rpc_client,
+        &block_reference,
+        account_id,
+    )
+    .await?;
+
+    println!();
+    let mut table = prettytable::Table::new();
+    table.set_format(*prettytable::format::consts::FORMAT_NO_COLSEP);
+
+    table.add_row(prettytable::row![
+        Fg->account_id,
+        format!("At block #{}\n({})", view_code_response.block_height, view_code_response.block_hash)
+    ]);
+
+    let contract_status = if account_view.code_hash == near_primitives::hash::CryptoHash::default()
+    {
+        "No contract code".to_string()
+    } else {
+        hex::encode(account_view.code_hash.as_ref())
+    };
+    table.add_row(prettytable::row![
+        Fy->"SHA-256 checksum hex",
+        contract_status
+    ]);
+
+    table.add_row(prettytable::row![
+        Fy->"Storage used",
+        format!("{} ({} Wasm + {} data)",
+            bytesize::ByteSize(account_view.storage_usage),
+            bytesize::ByteSize(u64::try_from(contract_code_view.code.len())?),
+            bytesize::ByteSize(
+                account_view.storage_usage
+                    .checked_sub(u64::try_from(contract_code_view.code.len())?)
+                    .expect("Unexpected error")
+            )
+        )
+    ]);
+
+    let access_keys_summary = if access_keys.is_empty() {
+        "Contract is locked (no access keys)".to_string()
+    } else {
+        let full_access_keys_count = access_keys
+            .iter()
+            .filter(|access_key| {
+                matches!(
+                    access_key.access_key.permission,
+                    near_primitives::views::AccessKeyPermissionView::FullAccess
+                )
+            })
+            .count();
+        format!(
+            "{} full access keys and {} function-call-only access keys",
+            full_access_keys_count,
+            access_keys.len() - full_access_keys_count
+        )
+    };
+    table.add_row(prettytable::row![
+        Fy->"Access keys",
+        access_keys_summary
+    ]);
+
+    if let Ok(contract_source_metadata_response) = get_contract_source_metadata(
+        &network_config.network_name,
+        &json_rpc_client,
+        &block_reference,
+        account_id,
+    )
+    .await
+    {
+        if let Ok(contract_source_metadata) = contract_source_metadata_response
+            .call_result()?
+            .parse_result_from_json::<self::contract_metadata::ContractSourceMetadata>(
```

Comment:
`get_contract_source_metadata` should better return `ContractSourceMetadata` type instead of the response type.

Reply:
Done